ext/datasource/datasource_test.go

```diff
@@ -63,3 +74,24 @@ func TestBase_indexOfHandler(t *testing.T) {
 		assert.True(t, b.indexOfHandler(h2) == 1, "Fail to execute the case TestBase_indexOfHandler.")
 	})
 }
+
+func TestBase_Handle(t *testing.T) {
+	t.Run("TestBase_handle", func(t *testing.T) {
+		b := &Base{
+			handlers: make([]PropertyHandler, 0),
+		}
+		h := NewSystemRulesHandler(SystemRuleJsonArrayParser)
+		b.handlers = append(b.handlers, h)
+		err := b.Handle([]byte(systemRule))
+		assert.Nil(t, err)
+	})
+	t.Run("TestBase_handleErr", func(t *testing.T) {
+		b := &Base{
+			handlers: make([]PropertyHandler, 0),
+		}
+		h := NewSystemRulesHandler(SystemRuleJsonArrayParser)
+		b.handlers = append(b.handlers, h)
+		err := b.Handle([]byte(systemRuleErr))
+		assert.NotNil(t, err)
+	})
+}
```

Comment:
Maybe better to add a case of multi handlers.

Reply:
updated